### PR TITLE
Created an API for managing the movie database

### LINKED ISSUE #3: Create Spring application with controller for python script usage

### PR DESCRIPTION
**Pull request by Mr-Keltt**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,6 +27,48 @@
...
+		</dependency>
+
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.projectlombok</groupId>
```
Comment (savin000) [inline pom.xml]: You added Lombok but don't actually use it. Why then?

File: src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @GetMapping("{id}")
+    public Movie getMovie(@PathVariable("id") Movie movie) {
+        return movie;
+    }
+
+    @PostMapping
+    public Movie create(@RequestBody Movie movie) {
+        System.out.println(movie); //todo create logs
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java]: remove this from final version that goes to main

File: src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import ru.vsu.csd.datatransferservice.domain.Movie;
+import ru.vsu.csd.datatransferservice.repos.MovieRepository;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("data_transfer")
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java]: don't use "_" in url. Use "-" instead

File: src/main/java/ru/vsu/csd/datatransferservice/domain/Movie.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
+package ru.vsu.csd.datatransferservice.domain;
+
+import jakarta.persistence.*;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.util.Random;
+
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/domain/Movie.java:15]: You can use Lombok annotations for constructor and setters/getters here if you want

File: src/main/java/ru/vsu/csd/datatransferservice/repos/MovieRepository.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package ru.vsu.csd.datatransferservice.repos;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import ru.vsu.csd.datatransferservice.domain.Movie;
+
+public interface MovieRepository extends JpaRepository<Movie, Long> {
+//    List<Movie> findByTitle(String title);
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/repos/MovieRepository.java]: remove comments

**Review by savin000 — COMMENTED** [review]
Check previous comments. Remember to remove unnecessary comments in production (main branch in our case)

File: pom.xml (hunk 1 above)
Comment (Mr-Keltt) [reply pom.xml]: I used it for the version with the interface and forgot to delete it.

File: .gitignore [hunk 6]
```diff
@@ -4,4 +4,7 @@ bin/
 !**/src/main/**/bin/
 .idea
 target/
-*.iml
\ No newline at end of file
+*.iml
+.mvn/
+.jython_cache/
+.node_modules/
```
Comment (savin000) [inline .gitignore]: we don't need to exclude node_modules from the project because we will not include any frontend logic here

File: pom.xml [hunk 7]
```diff
@@ -27,6 +27,41 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
```
Comment (savin000) [inline pom.xml:51]: You still have lombok in dependencies. Moreover, it is declared twice :)

File: pom.xml [hunk 8]
```diff
@@ -27,6 +27,41 @@
...
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>com.h2database</groupId>
```
Comment (savin000) [inline pom.xml]: Why do you have h2 dependency. We use postgres already. ANyway, we have to choose only one db

File: src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    @PutMapping("{id}")
+    public Movie update(@PathVariable("id") Movie movieFromDb, @RequestBody Movie movie) {
+        BeanUtils.copyProperties(movie, movieFromDb, "id");
+
+        return movieRepo.save(movieFromDb);
+    }
+
+    @DeleteMapping("{id}")
+    public void delete(@PathVariable("id") String id) {
+        if (id.equals("all")) {
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java]: just add delete mapping with no parameter id that deletes all movies. (It is a main REST API concept)

File: src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        return movieRepo.findAll();
+    }
+
+    @GetMapping("{id}")
+    public Movie getMovie(@PathVariable("id") Movie movie) {
+        return movie;
+    }
+
+    @PostMapping
+    public Movie create(@RequestBody Movie movie) {
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/controller/DataTransferController.java]: rename all methods to be like "getMovie", "deleteMovie", etc

File: src/main/java/ru/vsu/csd/datatransferservice/domain/Movie.java [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+@Entity
+@Table
+@ToString(of = {"id", "title"})
+@EqualsAndHashCode(of = "id")
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/domain/Movie.java:19]: Btw, here you use lombok

File: src/main/java/ru/vsu/csd/datatransferservice/domain/Movie.java [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+                emotionalColoringJoy,
+                emotionalColoringSadness,
+                emotionalColoringAnger,
+                emotionalColoringDisgust,
+                emotionalColoringFear};
+    }
+
+    public void setEmotionalColoring(double[] emotionalColoring) throws Exception {
+        if (emotionalColoring.length != 5) {
+            throw new Exception("incorrect number of elements in the array");
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/domain/Movie.java]: create custom exception

**Review by savin000 — COMMENTED** [review]
Check all comments

File: src/main/java/ru/vsu/csd/datatransferservice/service/TextAnalysisService.java [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+            // Запускаем процесс
+            Process process = pb.start();
+
+            // Читаем результат выполнения скрипта
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            ArrayList<String> line = new ArrayList<String>();
+            while (true) {
+                line.add(reader.readLine());
+                if (!(line.get(line.size() - 1) != null)) break;
```
Comment (savin000) [inline src/main/java/ru/vsu/csd/datatransferservice/service/TextAnalysisService.java]: I've seen this in another pr. Fix this please

**Review by savin000 — COMMENTED** [review]
remove changes from another pr

**Review by savin000 — APPROVED** [review]

**Pull request merged**